tests/vision_test.cpp

```diff
@@ -16,337 +11,173 @@
-            floor_above << ( above_cache.floor_cache[p.x][p.y] ? '#' : '.' ) << ' ';
-        }
-        fields << '\n';
-        transparency << '\n';
-        seen << '\n';
-        lm << '\n';
-        apparent_light << '\n';
-        obstructed << '\n';
-        floor_above << '\n';
-    }
+static const time_point midnight = calendar::turn_zero + 0_hours;
+static const time_point midday = calendar::turn_zero + 12_hours;
 
-    INFO( "zlevels: " << here.has_zlevels() );
-    INFO( "origin: " << origin );
-    INFO( "player: " << player_character.pos() );
-    INFO( "unimpaired_range: " << player_character.unimpaired_range() );
-    INFO( "vision_threshold: " << vvcache.vision_threshold );
-    INFO( "fields:\n" << fields.str() );
-    INFO( "transparency:\n" << transparency.str() );
-    INFO( "seen:\n" << seen.str() );
-    INFO( "lm:\n" << lm.str() );
-    INFO( "apparent_light:\n" << apparent_light.str() );
-    INFO( "obstructed:\n" << obstructed.str() );
-    INFO( "floor_above:\n" << floor_above.str() );
-
-    bool success = true;
-    std::ostringstream expected;
-    std::ostringstream observed;
-
-    for( int y = 0; y < height; ++y ) {
-        for( int x = 0; x < width; ++x ) {
-            const tripoint p = origin + point( x, y );
-            const lit_level level = here.apparent_light_at( p, vvcache );
-            const char exp_char = expected_results[y][x];
-            if( exp_char < '0' || exp_char > '9' ) {
-                FAIL( "unexpected result char '" <<
-                      expected_results[y][x] << "'" );
-            }
-            const int expected_level = exp_char - '0';
-
-            observed << level << ' ';
-            expected << expected_level << ' ';
-            if( level != expected_level ) {
-                success = false;
-            }
-        }
-        observed << '\n';
-        expected << '\n';
-    }
+static const move_mode_id move_mode_walk( "walk" );
+static const move_mode_id move_mode_crouch( "crouch" );
 
-    INFO( "observed:\n" << observed.str() );
-    INFO( "expected:\n" << expected.str() );
-    CHECK( success );
-}
+using namespace map_test_case_common;
+using namespace map_test_case_common::tiles;
+
+auto static const ter_set_flat_roof_above = ter_set( ter_str_id( "t_flat_roof" ), tripoint_above );
+
+static const std::function<bool( map_test_case::tile )> set_up_tiles_common =
+    ifchar( ' ', noop ) ||
+    ifchar( 'U', noop ) ||
+    ifchar( 'u', ter_set( ter_str_id( "t_floor" ) ) + ter_set_flat_roof_above ) ||
+    ifchar( 'L', ter_set( ter_str_id( "t_utility_light" ) ) + ter_set_flat_roof_above ) ||
+    ifchar( '#', ter_set( ter_str_id( "t_brick_wall" ) ) + ter_set_flat_roof_above ) ||
+    ifchar( '=', ter_set( ter_str_id( "t_window_frame" ) ) + ter_set_flat_roof_above ) ||
+    ifchar( '-', ter_set( ter_str_id( "t_floor" ) ) + ter_set_flat_roof_above ) ||
+    fail;
+
+struct vision_test_flags {
+    bool no_3d = false;
```

Comment:
Since you've turned this `enum` into a `struct` it might be clearer to invert this flag and have it be e.g. `test_3d` that defaults to `true`, thus avoiding double negatives?

Reply:
Update: another recent PR removed the need for "no 3D" tests entirely, so this struct member can just disappear.

Reply:
I removed the flag.